File: Host/robustpatchimpl/src/main/java/com/demokiller/robustpatchimpl/PatchesInfoImpl.java
package com.demokiller.robustpatchimpl;

import com.demokiller.robustpatch.PatchedClassInfo;
import com.demokiller.robustpatch.PatchesInfo;

import java.util.ArrayList;
import java.util.List;

public class PatchesInfoImpl implements PatchesInfo {
    public List<PatchedClassInfo> getPatchedClassesInfo() {
        List<PatchedClassInfo> patchedClassesInfos = new ArrayList<PatchedClassInfo>();
        PatchedClassInfo patchedClass = new PatchedClassInfo(
                "com.demokiller.host.MoneyBean",
                MoneyBeanStatePatch.class.getCanonicalName());
        patchedClassesInfos.add(patchedClass);
        return patchedClassesInfos;
    }
}

File: Host/robustpatchimpl/src/main/java/com/demokiller/robustpatchimpl/PatchesInfoImpl.kt
package com.demokiller.robustpatchimpl

import com.demokiller.robustpatch.PatchedClassInfo
import com.demokiller.robustpatch.PatchesInfo
import java.util.*

class PatchesInfoImpl : PatchesInfo {
    override fun getPatchedClassesInfo(): List<PatchedClassInfo?>? {
        val patchedClassesInfos: MutableList<PatchedClassInfo> = ArrayList()
        val patchedClass = PatchedClassInfo(
                "com.demokiller.host.MoneyBean",
                MoneyBeanStatePatch::class.java.canonicalName)
        patchedClassesInfos.add(patchedClass)
        return patchedClassesInfos
    }
}